lib/sidekiq/metrics/query.rb

```diff
@@ -35,48 +35,32 @@ def initialize(pool: Sidekiq.redis_pool, now: Time.now)
         @klass = nil
       end
 
-      # Get metric data from the last hour and roll it up
-      # into top processed count and execution time based on class.
-      def top_jobs
-        resultset = {}
-        resultset[:date] = @time.to_date
-        resultset[:period] = :hour
-        resultset[:ends_at] = @time
-        time = @time
+      # Get metric data for all jobs from the last hour
+      def top_jobs(minutes: 60)
+        result = Result.new
```

Comment:
I had a lot of trouble wrapping my head around the previous implementation and shaping the data the way I needed it for the chart. It helped me to extract the `Query::Result` and `Query::JobResult` classes here, but I'm sure we could also get it working with the nested Hashes if you don't like this approach.